cmd/src/batch_exec.go

```diff
@@ -174,6 +175,28 @@ func executeBatchSpecInWorkspaces(ctx context.Context, flags *executorModeFlags)
 		return errors.New("invalid execution, no steps to process")
 	}
 
+	// Pass the os.Environ to run steps to allow access to the secrets set
+	// in the executor environment.
+	// The executor runtime takes care of not forwarding any sensitive secrets
+	// from the host, so this is safe.
+	globalEnv := os.Environ()
+
+	// If the docker prefix registry is set, inject it into each step's container
+	// to fully qualify the container name
+	environ := map[string]string{}
+	for _, e := range globalEnv {
+		pair := strings.SplitN(e, "=", 2)
+		environ[pair[0]] = pair[1]
+	}
+
+	dockerPrefix, ok := environ["DOCKER_PREFIX_REGISTRY_URL"]
```

Comment:
could we instead make this a flag that the server passes? Or do we want to configure this per-executor?
That way we wouldn't need to parse the env here.